src/Microsoft.ML.StandardTrainers/IValueMapper.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.ML.Data
+{
+    [BestFriend]
+    internal delegate void ValueMapperEncrypted<TVal1, TVal2, TVal3>(in TVal1 val1, in TVal2 val2, ref TVal3 val3);
+
+    [BestFriend]
+    internal interface IValueMapperEncrypted : IValueMapper
+    {
+        ValueMapperEncrypted<TSrc, TKey, TDst> GetMapper<TSrc, TKey, TDst>();
+    }
```

Comment:
I would put this next to other mapper interfaces in machinelearning\src\Microsoft.ML.Core\Data\IValueMapper.cs. This is really one to two mapper, the name needs to be generic enough that others can also use this and should follow the pattern, so please remove "encrypted.".

Reply:
OK, but it will have to be named IValueMapperOneToTwo (or another name) to avoid a name conflict. And really, isn't it more a two inputs to one output mapper? Should it be IValueMapperTwoToOne?

Reply:
sorry, yes, Two to One, that is what I had meant :)